### PR TITLE
add transfer

### PR DESCRIPTION
**Pull request by Leopoldthecoder**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by Leopoldthecoder** [comment]
Pending English doc and tests.

File: packages/transfer/src/main.vue [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    <div class="el-transfer__buttons">
+      <el-button
+        type="primary"
+        size="small"
+        @click.native="addToLeft"
+        :disabled="rightChecked.length === 0">
+        <i class="el-icon-arrow-left"></i>
+        <span v-if="buttonTexts[0] !== undefined">{{ buttonTexts[0] }}</span>
+      </el-button>
+      <el-button
```
Comment (QingWei-Li) [inline packages/transfer/src/main.vue]: 用到 button 应该把 button 引入

File: packages/transfer/src/transfer-panel.vue [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    components: {
+      ElCheckboxGroup,
+      ElCheckbox,
+      OptionContent: {
+        props: {
+          option: Object
+        },
+        render(h) {
+          const getParent = vm => {
+            if (vm.$options.name === 'ElTransferPanel') {
```
Comment (QingWei-Li) [inline packages/transfer/src/transfer-panel.vue]: 要自定义一个属性，name 的话外面是可以改的

File: packages/transfer/src/main.vue [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      },
+      buttonTexts: {
+        type: Array,
+        default() {
+          return [];
+        }
+      },
+      filterPlaceholder: {
+        type: String,
+        default: '请输入搜索内容'
```
Comment (Leopoldthecoder) [inline packages/transfer/src/main.vue]: 没有办法做 i18n 了，因为其他二十多种语言的 `请输入搜索内容` 不知道怎么说，把贡献者拉过来补这句也不现实。

File: packages/transfer/src/main.vue (hunk 1 above)
Comment (Leopoldthecoder) [inline packages/transfer/src/main.vue]: 大佬教育得是

File: packages/transfer/src/transfer-panel.vue (hunk 2 above)
Comment (Leopoldthecoder) [inline packages/transfer/src/transfer-panel.vue]: 大佬教育得是

File: packages/transfer/src/main.vue (hunk 3 above)
Comment (QingWei-Li) [inline packages/transfer/src/main.vue]: 你的机器人呢

Comment (Leopoldthecoder) [inline packages/transfer/src/main.vue]: 别闹

Comment (QingWei-Li) [inline packages/transfer/src/main.vue]: 感觉没毛病 其他语言的都用英文，都老外自己来提 PR

**Pull request closed without merging**